lib/coll.gi

```diff
@@ -169,8 +169,11 @@ InstallMethod( IsWholeFamily,
 ##
 #M  Size( <C> ) . . . . . . . . . . . . . . . . . . . .  size of a collection
 ##
-InstallImmediateMethod( Size,
-    IsCollection and HasIsFinite and IsAttributeStoringRep, 0,
+#  replace immediate method by ordinary, as immediate gets called for every
+#  group that knows it is finite but does not know its size -- e.g.
+#  permutation, pc.  The benefit of this is minimal beyond showing off a
+#  feature.
+InstallMethod( Size,true, [IsCollection and HasIsFinite], 0,
```

Comment:
You can omit true and 0 here

Reply:
OK, but is it a problem if they're in? (Some methods actually might need a higher ranking).